examples/neural_compressor/text-generation/run_generation.py

```diff
@@ -0,0 +1,320 @@
+        action="store_true",
+        help="Whether to use 16-bit (mixed) precision (through NVIDIA apex) instead of 32-bit",
+    )
+    parser.add_argument("--jit", action="store_true", help="Whether or not to use jit trace to accelerate inference")
+    args = parser.parse_args()
+
+    args.device = torch.device("cuda" if torch.cuda.is_available() and not args.no_cuda else "cpu")
+    args.n_gpu = 0 if args.no_cuda else torch.cuda.device_count()
+
+    logger.warning(f"device: {args.device}, n_gpu: {args.n_gpu}, 16-bits training: {args.fp16}")
+
+    set_seed(args)
+
+    # Initialize the model and tokenizer
+    try:
+        args.model_type = args.model_type.lower()
+        model_class, tokenizer_class = MODEL_CLASSES[args.model_type]
+    except KeyError:
+        raise KeyError("the model {} you specified is not supported. You are welcome to add it and open a PR :)")
+
+    tokenizer = tokenizer_class.from_pretrained(args.model_name_or_path)
+    # model = model_class.from_pretrained(args.model_name_or_path)
+    model = AutoModelForGeneration.from_pretrained(args.model_name_or_path)
+    model.to(args.device)
+
+    if args.fp16:
+        model.half()
+
+    args.length = adjust_length_to_model(args.length, max_sequence_length=model.config.max_position_embeddings)
+    logger.info(args)
+
+    prompt_text = args.prompt if args.prompt else input("Model prompt >>> ")
+
+    # Different models need different input formatting and/or extra arguments
+    requires_preprocessing = args.model_type in PREPROCESSING_FUNCTIONS.keys()
+    if requires_preprocessing:
+        prepare_input = PREPROCESSING_FUNCTIONS.get(args.model_type)
+        preprocessed_prompt_text = prepare_input(args, model, tokenizer, prompt_text)
+
+        if model.__class__.__name__ in ["TransfoXLLMHeadModel"]:
+            tokenizer_kwargs = {"add_space_before_punct_symbol": True}
+        else:
+            tokenizer_kwargs = {}
+
+        encoded_prompt = tokenizer.encode(
+            preprocessed_prompt_text, add_special_tokens=False, return_tensors="pt", **tokenizer_kwargs
+        )
+    else:
+        prefix = args.prefix if args.prefix else args.padding_text
+        encoded_prompt = tokenizer.encode(prefix + prompt_text, add_special_tokens=False, return_tensors="pt")
+    encoded_prompt = encoded_prompt.to(args.device)
+
+    if encoded_prompt.size()[-1] == 0:
+        input_ids = None
+    else:
+        input_ids = encoded_prompt
+
+    if args.jit:
+        from optimum.exporters.tasks import TasksManager
+
+        model.config.return_dict = False
+        onnx_config_class = TasksManager.get_exporter_config_constructor(
+            exporter="onnx", model=model, task=GENERATION_TASK[args.model_type]
+        )
+        onnx_config = onnx_config_class(model.config)
+        dummy_inputs = onnx_config.generate_dummy_inputs(framework="pt")
+        inputs_tuple = (dummy_inputs["input_ids"], dummy_inputs["past_key_values"], dummy_inputs["attention_mask"])
```

Comment:
The arguments position might vary for different model architecture, when creating `inputs_tuple` the signature of the forward method of the original model can be used as done for the OVTrainer ONNX export, see https://github.com/huggingface/optimum-intel/blob/v1.7.3/optimum/intel/openvino/trainer.py#L780

Reply:
Will remove if we fixed dummy_inputs issue.

Reply:
Removed